packages/amplify_auth_cognito/example/ios/amplify_auth_cognito_tests/amplify_auth_cognito_tests.swift

```diff
@@ -0,0 +1,1168 @@
+                let authError = AuthError.service("Invalid username", "Enter a username", nil)
+                handleAuthError(error: authError, flutterResult: flutterResult, msg: FlutterAuthErrorMessage.CONFIRM_SIGNUP.rawValue)
+            }
+        }
+        
+        plugin = SwiftAuthCognito.init(cognito: ConfirmSignUpMock())
+        
+        _data = [
+            "username": _username,
+            "confirmationCode": _confirmationCode,
+        ]
+        _args = ["data": _data]
+        
+        let call = FlutterMethodCall(methodName: "confirmSignUp", arguments: _args)
+        plugin.handle(call, result: {(result)->Void in
+            if let res = result as? FlutterError {
+                XCTAssertEqual( "AmplifyException", res.code )
+                XCTAssertEqual( "AMPLIFY_CONFIRM_SIGNUP_FAILED", res.message)
+            } else {
+                XCTFail()
+            }
+        })
+    }
+    
+    func test_confirmSignUpValidation() {
+        let rawOptions: Dictionary<String, Any> = ["foo": "bar"]
+        var rawData: NSMutableDictionary = ["username": _username]
+        let failWithNoConfirmationCode = FlutterConfirmSignUpRequest.validate(dict: rawData);
+        XCTAssertEqual(false, failWithNoConfirmationCode)
+        rawData = ["confirmationCode": _confirmationCode]
+        let failWithNoUsername = FlutterConfirmSignUpRequest.validate(dict: rawData);
+        XCTAssertEqual(false, failWithNoUsername)
+        rawData = ["options": rawOptions]
+        let succeedWithOptions = FlutterConfirmSignUpRequest.validate(dict: rawData);
+        XCTAssertEqual(true, succeedWithOptions)
+    }
+    
+    func test_resendSignUpCodeSuccessEmail() {
+        
+        class ResendSignUpCodeMock: AuthCognitoBridge {
+            override func onResendSignUpCode(flutterResult: @escaping FlutterResult, request: FlutterResendSignUpCodeRequest) {
+                  let resendCodeRes = Result<AuthCodeDeliveryDetails, AuthError>.success(AuthCodeDeliveryDetails(destination: DeliveryDestination.email(_email)))
+                  let resendData = FlutterResendSignUpCodeResult(res: resendCodeRes)
+                  flutterResult(resendData)
+            }
+        }
+        
+        plugin = SwiftAuthCognito.init(cognito: ResendSignUpCodeMock())
+        
+        _data = ["username": _username]
+        _args = ["data": _data]
+        let call = FlutterMethodCall(methodName: "resendSignUpCode", arguments: _args)
+        plugin.handle(call, result: {(result)->Void in
+            if let res = result as? FlutterResendSignUpCodeResult {
+                XCTAssertEqual( _email,  res.codeDeliveryDetails["destination"] )
+                XCTAssertEqual( "email",  res.codeDeliveryDetails["attributeName"] )
+                XCTAssertEqual( _email, (res.toJSON()["codeDeliveryDetails"] as! [String: String])["destination"])
```

Comment:
Is there any difference between 1st and 3rd assertion here?

Reply:
Yeah, the 3rd is testing the toJSON method that is used for serialization.